src/main/java/com/rultor/profiles/ProfileDeprecationAware.java

```diff
@@ -0,0 +1,125 @@
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.rultor.profiles;
+
+import com.jcabi.log.Logger;
+import com.jcabi.xml.XML;
+import com.rultor.spi.Profile;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Decorator allowing to add a deprecation message if and only if the
+ * {@link Profile} is deprecated.
+ *
+ * @author Nicolas Filotto (nicolas.filotto@gmail.com)
+ * @version $Id$
+ * @since 1.59
+ */
+public final class ProfileDeprecationAware implements Profile {
+
+    /**
+     * The underlying profile.
+     */
+    private final transient Profile profile;
+
+    /**
+     * Constructs a {@code ProfileDeprecationAware} with the specified
+     * underlying {@code Profile}.
+     * @param prof The underlying profile.
+     */
+    public ProfileDeprecationAware(final Profile prof) {
+        this.profile = prof;
+    }
+
+    @Override
+    public String name() {
+        return this.profile.name();
+    }
+
+    @Override
+    public XML read() throws IOException {
+        final XML xml = this.profile.read();
+        this.check(xml);
+        return xml;
+    }
+
+    @Override
+    public Map<String, InputStream> assets() throws IOException {
+        return this.profile.assets();
+    }
+
+    /**
+     * Checks if the profile is deprecated and if so log the deprecation notice.
+     * @throws IOException if it fails while getting the XML format of the
+     *  profile
+     */
+    public void check() throws IOException {
+        this.check(this.profile.read());
+    }
+
+    /**
+     * Indicates whether the profile is deprecated or not.
+     * @param prof The XML representation of the profile to test.
+     * @return True if the profile is deprecated, false otherwise.
+     */
+    public static boolean deprecated(final XML prof) {
+        final List<XML> images = prof.nodes(
+            "/p/entry[@key='docker']/entry[@key='image']"
+        );
+        boolean deprecated = false;
+        if (images.isEmpty()) {
+            deprecated = true;
+        } else {
+            for (final XML image : images) {
+                if ("yegor256/rultor".equals(image.node().getTextContent())) {
+                    deprecated = true;
+                    break;
+                }
+            }
+        }
+        return deprecated;
+    }
+
+    /**
+     * Checks if the profile is deprecated and if so log the deprecation notice.
+     * @param prof The XML representation of the profile to check.
+     */
+    private void check(final XML prof) {
+        if (ProfileDeprecationAware.deprecated(prof)) {
+            Logger.warn(
+                this,
+                "#### Deprecation Notice #### Don't use the default image"
```

Comment:
@essobedo can we provide more description here? Originally @sebing proposed couple of sentences for this notice